shared-data/python/setup.py

```diff
@@ -18,30 +22,44 @@
 
 DATA_ROOT = ".."
 DATA_SUBDIRS = ["deck", "labware", "module", "pipette", "protocol"]
+DATA_TYPES = ["definitions", "schemas"]
 DEST_BASE_PATH = "data"
 
 
-def get_shared_data_files():
+def get_shared_data_files() -> List[Path]:
     to_include = []
+
     for subdir in DATA_SUBDIRS:
-        top = os.path.join(DATA_ROOT, subdir)
-        for dirpath, dirnames, filenames in os.walk(top):
-            from_source = os.path.relpath(dirpath, DATA_ROOT)
-            to_include.extend([os.path.join(from_source, fname) for fname in filenames])
+        for data_type in DATA_TYPES:
+            data_dir = Path(DATA_ROOT) / subdir / data_type
+            if data_dir.is_dir():
+                to_include.extend(data_dir.glob("**/*.json"))
+
     return to_include
 
 
+def _minimize_and_write_json(data_file: Path, target_file: Path) -> None:
+    contents = json.dumps(json.loads(data_file.read_text()))
```

Comment:
```suggestion
    contents = json.dumps(json.loads(data_file.read_text()), separators=(',', ':'))
```
from the [compact encoding example of the json docs](https://docs.python.org/3/library/json.html) - the default encoding puts spaces after separators, this suppresses that